### PR TITLE
refactor: evm ecdsa adjustments

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary
https://app.clickup.com/t/86dv4y6kn

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: src/Transactions/Deserializer.php [hunk 1]
```diff
@@ -108,84 +113,72 @@ private function decodePayload(array $data): ?array
...
+        //             } else {
+        //                 throw new \Exception('Duplicate participant in multisignature');
+        //             }
+
+        //             $data['signatures'][] = $multiSignaturePart;
+        //         }
+        //     } else {
+        //         throw new \Exception('signature buffer not exhausted');
+        //     }
+        // }
```
Comment (ItsANameToo) [inline src/Transactions/Deserializer.php]: would just remove this if not needed right now

File: src/Transactions/Serializer.php [hunk 2]
```diff
@@ -119,20 +81,39 @@ private function serializeData(ByteBuffer $buffer, array $options = []): void
...
-        $buffer->writeUint256($this->transaction->data['fee']);
+        // if (! $skipSecondSignature) {
+        //     if (isset($this->transaction->data['secondSignature'])) {
+        //         $buffer->writeHex($this->transaction->data['secondSignature']);
+        //     }
+        // }
+
+        // if (! $skipMultiSignature && isset($this->transaction->data['signatures'])) {
+        //     $buffer->writeHex(implode('', $this->transaction->data['signatures']));
+        // }
```
Comment (ItsANameToo) [inline src/Transactions/Serializer.php]: can remove, unsure how multisig is going to work as it requires a contract for that

File: src/Transactions/Types/AbstractTransaction.php [hunk 3]
```diff
@@ -55,14 +65,34 @@ public function getBytes($options = []): Buffer
...
+
+        // Extract the recovery ID (an integer between 0 and 3) from the signature
+        $recoveryId = $signature->getRecoveryId();
+
+        // Get the full signature buffer, which includes the adjusted recovery ID at the start
+        $signatureHexWithRecoveryId = $signature->getBuffer()->getHex();
 
-        // $this->data['signature'] = $keys->sign($transaction)->getBuffer()->getHex();
-        $this->data['signature'] = $this->temporarySignerSign($transaction, $keys);
+        // Apparently, the compact signature returned by signCompact() includes an adjusted recovery ID
```
Comment (ItsANameToo) [inline src/Transactions/Types/AbstractTransaction.php:76]: good comment

File: src/Transactions/Builder/AbstractTransactionBuilder.php [hunk 4]
```diff
@@ -20,20 +17,13 @@ public function __construct(?array $data = null)
...
-                    'gasLimit' => 1000000,  // Default gas limit
-                    'payload'  => '',       // EVM code in hex format
-                ],
-            ],
+            'value'             => '0',
+            'senderPublicKey'   => '',
+            'gasPrice'          => '5',
+            'nonce'             => '1',
+            'network'           => Network::get()->pubKeyHash(),
+            'gasLimit'          => 1_000_000,
```
Comment (ItsANameToo) [inline src/Transactions/Builder/AbstractTransactionBuilder.php:25]: can we set default gas limits for the supported transaction types? e.g.:
- transfer: 21000
- validator registration: 500000
- validator resignation: 150000
- vote: 200000
- unvote: 200000
- default: 1000000

I didn't see them being overwritten in the code in the builders, but maybe I overlooked something

File: src/Enums/Fees.php [hunk 5]
```diff
@@ -22,5 +22,5 @@ class Fees
 
     public const USERNAME_RESIGNATION = '2500000000';
```
Comment (ItsANameToo) [inline src/Enums/Fees.php]: all the fees here are unused now I assume? 🤔 since it's all based on `gas_price` and `gas_limit` to determine the actual fee

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: src/Transactions/Builder/AbstractTransactionBuilder.php (hunk 4 above)
Comment (alfonsobries) [reply src/Transactions/Builder/AbstractTransactionBuilder.php:25]: @ItsANameToo Deleted all the code related to fees, it's not actually being used anywhere

**Comment by alfonsobries** [comment]
@ItsANameToo ready

**Comment by alfonsobries** [comment]
im also removing the Fee section on the docs here https://github.com/ArdentHQ/docs-poc-protocol/pull/74

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**